feat(navbar): create Navbar component

eddybrando opened the pull request:
This PR creates a reusable **Navbar** component.

We also create and export components for some icons and the Qiskit logo (needed to solve an issue with Platypus / Qiskit.org compatibility. Might be changed for SVGs after Nuxt 3 beta launches).

**Note:** You can see the preview in the latest comment with a _Storybook preview_ link.

---

Closes https://github.com/qiskit-community/qiskit-vue/issues/25

Closes Qiskit/platypus#454

eddybrando committed:
feat(navbar): create component skeleton

eddybrando committed:
feat(navbar): display logo and links

eddybrando committed:
feat(navbar): display links and dropdowns dynamically

eddybrando committed:
feat(navbar): style items

eddybrando committed:
feat(navbar): style dropdown

eddybrando committed:
feat(navbar): toggle menu

eddybrando committed:
feat(navbar): style collapsable menu

eddybrando committed:
feat(navbar): small screen style adjustments

eddybrando committed:
feat(navbar): link nav items

eddybrando committed:
feat(navbar): full height on small screens

eddybrando committed:
feat(navbar): medium size screen design

eddybrando committed:
feat(logo): create logo component

eddybrando committed:
feat(navbar): toggle icons

eddybrando committed:
feat(navbar): logo color

eddybrando committed:
feat(navbar): style corrections

eddybrando committed:
feat(navbar): style corrections

eddybrando committed:
feat(navbar): footer for small screens

eddybrando committed:
feat(icons): export icon components

eddybrando committed:
feat(navbar): bottom border

eddybrando committed:
feat(navbar): bottom border

eddybrando committed:
chore: remove unused file

eddybrando committed:
ci: link Storybook preview

eddybrando committed:
ci: link to Storybook preview

eddybrando committed:
ci: link to Storybook preview with correct SHA

eddybrando committed:
ci: link to Storybook preview with short SHA

eddybrando committed:
chore: remove unnecessary configuration

abdonrd commented on src/components/navbar/Navbar.vue:
```diff
@@ -0,0 +1,520 @@
+        </li>
+      </ul>
+
+      <footer class="navbar__footer">
+        <div class="navbar__footer__social-links">
+          <h4 class="navbar__footer__social-links__title">Stay Connected</h4>
+          <div class="navbar__footer__social-links__icons">
+            <a
+              v-for="{ icon, label, url } in socialMediaLinks"
+              :key="label"
+              class="navbar__footer__social-links__icons__icon"
+              :href="url"
+              rel="noopener"
+              target="_blank"
+              :title="label"
+            >
+              <component :is="icon" />
+            </a>
+          </div>
+        </div>
+        <div class="navbar__footer__notice">© Qiskit | All Rights Reserved</div>
+      </footer>
```
Couldn't the footer be undocked from the navbar? Thoughts?

eddybrando replied:
Just in case, this is not the _page's footer_, but the footer that appears in the navbar in small screens.

abdonrd reviewed (commented):
Why SVG logo and icons as components and not as `.svg` files?

And use it like `<img src="logo.svg">`

eddybrando commented on .storybook/preview.js:
```diff
@@ -1,3 +1,4 @@
+import "carbon-web-components/es/components/dropdown";
```
It caused issues with Nuxt's SSR on qiskit.org that broke the page.

eddybrando replied:
There was a similar issue with SVG images on Platypus. They didn't want to load.

This might be related to the way we currently ship the components (without compiling / packaging the components / assets). Once Nuxt 3 is available and we can drop support for Vue 2, I think that we can start to ship the components properly as intended by `vue-cli`. And I hope to then be able to replaces the SVG components with proper SVG files.

eddybrando replied:
Exactly, so we need to also include the web component in the project that uses the Navbar.

This is hopefully only a provisory solution until we can properly package components after Nuxt 3 is available.

eddybrando committed:
docs: add todo to import Carbon's dropdown

Related to #31

eddybrando committed:
refactor(basic-link): move style to class

As proposed in https://github.com/qiskit-community/qiskit-vue/pull/28#discussion_r723996731

eddybrando committed:
refactor(basic-link): move style to class
